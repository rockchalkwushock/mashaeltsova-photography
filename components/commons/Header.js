import { styles } from '../../lib'

const Header = ({ children }) => (
  <header>
    {children}
    <style jsx>{`
      header {
        align-items: center;
        background-color: ${styles.colors.peach};
        display: flex;
        flex-direction: column;
        justify-content: center;
        height: 100vh;
        padding: 1em;
        text-align: center;
      }
    `}</style>
  </header>
)

export default Header
