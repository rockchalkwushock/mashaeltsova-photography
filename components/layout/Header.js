import PropTypes from 'prop-types'

import { styles } from '../../lib'

/**
 * TODO FIXME
 *
 * Will want to render a different view for header when at
 * `/gallery` vs `/` because the gallery will already be heavy
 * with all the images being loaded.
 *
 * Pass `pathname` on the header and then run and if/else.
 * Possibly render a banner with the logo in one corner and
 * 'Masha Eltsova Photography' spanning the rest in Courgette or Nunito.
 */

const Header = ({ children, className }) => (
  <header className={`header ${className}`}>
    {children}
    <style jsx>{`
      header {
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
        background-color: ${styles.colors.peach};
        display: -webkit-box;
        display: -ms-flexbox;
        display: flex;
        -webkit-box-orient: vertical;
        -webkit-box-direction: normal;
        -ms-flex-direction: column;
        flex-direction: column;
        -webkit-box-pack: center;
        -ms-flex-pack: center;
        justify-content: center;
        padding: 1em;
        text-align: center;
      }
      .index {
        height: 95vh;
      }
      .gallery {
        height: 25vh;
      }
      @media (orientation: landscape) {
        .gallery {
          height: 50vh;
        }
      }
    `}</style>
  </header>
)

Header.propTypes = {
  className: PropTypes.string.isRequired
}

export default Header
