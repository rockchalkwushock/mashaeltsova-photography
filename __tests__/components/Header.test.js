import React from 'react'
import { render } from 'enzyme'

import Header from '../../components/commons/Header'

describe('Component: Header', () => {
  test('Renders without exploding from /', () => {
    const tree = render(<Header className="index" />)
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding from /gallery', () => {
    const tree = render(<Header className="gallery" />)
    expect(tree).toMatchSnapshot()
  })
})
