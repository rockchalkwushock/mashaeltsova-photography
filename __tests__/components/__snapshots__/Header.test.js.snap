// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header Renders without exploding from / 1`] = `
<header
  class="jsx-3692150445 header index"
/>
`;

exports[`Component: Header Renders without exploding from /gallery 1`] = `
<header
  class="jsx-3692150445 header gallery"
/>
`;
